src/wp-includes/default-filters.php

```diff
@@ -752,4 +752,29 @@
 add_action( 'before_delete_post', '_wp_before_delete_font_face', 10, 2 );
 add_action( 'init', '_wp_register_default_font_collections' );
 
+// TODO: This should probably go somewhere else.
+function set_ignored_hooked_blocks_metadata_upon_rest_insert( $post ) {
+	$hooked_blocks = get_hooked_blocks();
+	if ( empty( $hooked_blocks ) && ! has_filter( 'hooked_block_types' ) ) {
+		return;
+	}
+
+	// At this point, the post has already been created.
+	// We need to build the corresponding `WP_Block_Template` object.
+	// To that end, we need to prevent hooked blocks insertion from running again.
+
+	// Suppress all hooked blocks getting inserted into the context.
+	add_filter( 'hooked_block_types', '__return_empty_array', 99999, 0 );
+	$template = _build_block_template_result_from_post( $post );
+	remove_filter( 'hooked_block_types', '__return_empty_array', 99999 );
+
+	$before_block_visitor = make_before_block_visitor( $hooked_blocks, $template, 'set_ignored_hooked_blocks_metadata' );
+	$after_block_visitor  = make_after_block_visitor( $hooked_blocks, $template, 'set_ignored_hooked_blocks_metadata' );
+
+	$blocks        = parse_blocks( $template->content );
+	$post->content = traverse_and_serialize_blocks( $blocks, $before_block_visitor, $after_block_visitor );
+}
+add_action( 'rest_insert_wp_template', 'set_ignored_hooked_blocks_metadata_upon_rest_insert', 10, 1 );
+// Do we need to add it separately for wp_template_part?
```

Comment:
I _believe_ so since they are a different post type `wp_template_part`

Reply:
Done in https://github.com/WordPress/wordpress-develop/pull/6087/commits/c0b78ad832b456c84a58dfb118afd2eb1eb1210d 👍